jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/FilterDiagnosticsCollector.java

```diff
@@ -1,134 +1,105 @@
+    public FilterDiagnosticsCollector() {
+        super();
+    }
+
+    @Override
+    protected String getDiagnosticSource() {
+        return ServletConstants.DIAGNOSTIC_SOURCE;
+    }
+
+    public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnostics) {
+        if (unit != null) {
+            IType[] alltypes;
+            IAnnotation[] allAnnotations;
+
+            try {
+                alltypes = unit.getAllTypes();
+                for (IType type : alltypes) {
+                    allAnnotations = type.getAnnotations();
+                    IAnnotation webFilterAnnotation = null;
+
+                    for (IAnnotation annotation : allAnnotations) {
+                        if (isMatchedJavaElement(type, annotation.getElementName(),
+                                ServletConstants.WEBFILTER_FQ_NAME)) {
+                            webFilterAnnotation = annotation;
+                        }
+                    }
+
+                    String[] interfaces = { ServletConstants.FILTER_FQ_NAME };
+                    boolean isFilterImplemented = doesImplementInterfaces(type, interfaces);
+
+                    if (webFilterAnnotation != null && !isFilterImplemented) {
+                        diagnostics.add(createDiagnostic(type, unit,
+                                "Annotated classes with @WebFilter must implement the Filter interface.",
+                                ServletConstants.DIAGNOSTIC_CODE_FILTER, null, DiagnosticSeverity.Error));
+                    }
+
+                    /* URL pattern diagnostic check */
+                    if (webFilterAnnotation != null) {
+                        IMemberValuePair[] memberValues = webFilterAnnotation.getMemberValuePairs();
+
+                        boolean isUrlpatternSpecified = false;
+                        boolean isServletNamesSpecified = false;
+                        boolean isValueSpecified = false;
+                        for (IMemberValuePair mv : memberValues) {
+                            if (mv.getMemberName().equals(ServletConstants.URL_PATTERNS)) {
+                                isUrlpatternSpecified = true;
+                                continue;
+                            }
+                            if (mv.getMemberName().equals(ServletConstants.SERVLET_NAMES)) {
+                                isServletNamesSpecified = true;
+                                continue;
+                            }
+                            if (mv.getMemberName().equals(ServletConstants.VALUE)) {
+                                isValueSpecified = true;
+                            }
+                        }
+                        if (!isUrlpatternSpecified && !isValueSpecified && !isServletNamesSpecified) {
+                            diagnostics.add(createDiagnostic(webFilterAnnotation, unit,
+                                    "The annotation @WebServlet must define the attribute 'urlPatterns', 'servletNames' or 'value'.",

```

Comment:
This should say `The annotation @WebFilter...` per https://github.com/eclipse/lsp4jakarta/pull/296#discussion_r946821684